server/src/test/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinatorTest.java

```diff
@@ -3192,4 +3192,59 @@ public void testTimelineWith1CorePartitionTombstone() throws IOException
     SegmentTimeline segmentTimeline = SegmentTimeline.forSegments(allUsedSegments);
     Assert.assertEquals(0, segmentTimeline.lookup(interval).size());
   }
+
+  @Test
+  public void testSegmentIdMustNotBeReused() throws IOException
```

Comment:
This test should probably live in `SegmentAllocateActionTest` next to the sister test `testSegmentIsAllocatedForLatestUsedSegmentVersion`. It should also be renamed to include some more info, such as
`testSegmentIsNotAllocatedForUnusedSegmentVersion`.

Reply:
Done.
Did not rename the method however since the version is not necessarily unused. It can occur if any unused or used segment conflicts with an allocated pending segment id.